astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,271 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This module contains convenience functions for retrieving solar system
+ephemerides from jplephem.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+from ..extern import six
+from .sky_coordinate import SkyCoord
+from ..utils.data import download_file
+from .. import units as u
+from ..constants import c as speed_of_light
+from .representation import CartesianRepresentation
+from .builtin_frames import GCRS, ICRS
+from .builtin_frames.utils import get_jd12, cartrepr_from_matmul
+from .. import _erfa
+
+__all__ = ["get_body", "get_moon", "get_barycentric_body_position", "SOLAR_SYSTEM_BODIES"]
+
+KERNEL = None
+
+"""
+each value in the BODIES dictionary a list of kernel pairs needed
+to find the barycentric position of that object from the JPL kernel.
+"""
+SOLAR_SYSTEM_BODIES = ['sun', 'mercury', 'venus', 'earth-moon-barycenter',
+                       'earth', 'moon', 'mars', 'jupiter', 'saturn', 'uranus', 
+                       'neptune', 'pluto']
+BODIES = {'sun': [(0, 10)],
+          'mercury': [(0, 1), (1, 199)],
+          'venus': [(0, 2), (2, 299)],
+          'earth-moon-barycenter': [(0, 3)],
+          'earth':  [(0, 3), (3, 399)],
+          'moon': [(0, 3), (3, 301)],
+          'mars': [(0, 4)],
+          'jupiter': [(0, 5)],
+          'saturn': [(0, 6)],
+          'uranus': [(0, 7)],
+          'neptune': [(0, 8)],
+          'pluto': [(0, 9)]}
+
+
+def _download_spk_file(show_progress=True):
+    """
+    Get the Satellite Planet Kernel (SPK) file `de430.bsp` from NASA JPL.
+
+    Download the file from the JPL webpage once and subsequently access a
+    cached copy. This file is ~120 MB, and covers years ~1550-2650 CE [1]_.
+
+    .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
+    """
+    de430_url = ('http://naif.jpl.nasa.gov/pub/naif/'
```

Comment:
Just to generalize a little, maybe put this as an argument to the routine, i.e.,
```
def _download_spk_file(url='http://....', show_progress=True):
   ...
```
(not at all essential, though)